figures/02_parameter_table.Rmd

```diff
@@ -0,0 +1,193 @@
+  rn[[i]] <- dat
+}
+
+# Column names still uneven
+cnames <- Reduce(intersect, c(lapply(hn, colnames), 
+                              lapply(rn, colnames)))
+keep_match <- function(x) {x[, colnames(x) %in% cnames]}
+hn <- lapply(hn, keep_match)
+rn <- lapply(rn, keep_match)
+all_df <- rbind.data.frame(do.call(rbind, hn), do.call(rbind, rn))
+
+# Set CO2 chamber conditions, match to actual Ca
+clevels <- c(400, 300, 200, 100, 50, 20, 10, 
+             500, 600, 700, 800, 900, 1000, 1200, 1500)
+nearest_clevel <- Vectorize(function(x) {clevels[which.min(abs(x - clevels))]})
+
+# set Qin chamber conditions, match to actual Qin
+qlevels <- c(0, 50, 100, 200, 400, 
+             600, 800, 1200, 1500)
+nearest_qlevel <- Vectorize(function(x) {qlevels[which.min(abs(x - qlevels))]})
+
+# Keep relevant columns
+all <- all_df %>%
+  select(Treatment, trt, ID, species, rep, obs, date, E, A, Ca, Ci, Pci, Pca, Qin, gsw) %>%
+  mutate(roundCa = nearest_clevel(Ca),
+         roundQin = nearest_qlevel(Qin))
+
+all <- as.data.table(all)
+
+# Functions to summarize
+SE <- function(x) {sd(x, na.rm = T)/sqrt(sum(!is.na(x)))}
+num <- function(x) {sum(!is.na(x))}
+
+# Summarize A and gs for only when Ca = 400 and Qin = 1500, first by plant, then by treatment
+all_sum <- all[roundCa == 400 & roundQin == 1500,
+               .(A.m = mean(A,  na.rm = TRUE), A.s = SE(A), A.n = num(A),
+                 gsw.m = mean(gsw, na.rm = TRUE), gsw.s = SE(gsw), gsw.n = num(gsw)),
+               by = .(trt, ID, rep)]
+all_sum2 <- all_sum[,.(A.m = mean(A.m,  na.rm = TRUE), A.s = SE(A.m), A.n = num(A.m),
+                      gsw.m = mean(gsw.m, na.rm = TRUE), gsw.s = SE(gsw.m), gsw.n = num(gsw.m)),
+                    by = .(trt)]
+
+# Reformat from wide to narrow to wide again
+ag_sum <- all_sum2 %>%
+  pivot_longer(!trt, names_to = c("trait", "value."), names_pattern = "(.*).(.)") %>%
+  pivot_wider(names_from = c("value."), values_from = c("value")) %>%
+  rename(p.mean = m, p.se = s, p.n = n) %>%
+  pivot_wider(names_from = trt, values_from = c(p.mean, p.se, p.n), names_sep = ".") 
+  
+```
+
+Import parameters for all experiments and summarize. 
+
+```{r}
+in_params <- as.data.table(read.csv("../../sentinel-detection/data/derived_data/parameters_data.csv"))
+
+params_sum <- in_params[Treatment %in% c("31_22_250", "31_31_250"), .(p.mean = mean(Value, na.rm = T), 
+                            p.se = SE(Value),
+                            p.n = num(Value)), by = .(Treatment, trait)]
+p_sum <- params_sum %>%
+  mutate(trt = case_when(Treatment == "31_22_250" ~ "Control",
+                         Treatment == "31_31_250" ~ "High night T")) %>%
+  select(-Treatment) %>%
+  pivot_wider(names_from = trt, values_from = c(p.mean, p.se, p.n), names_sep = ".") %>%
+  filter(trait != "g0M", trait != "g1M") %>%
+  bind_rows(ag_sum) %>%
+  mutate(across(2:5, round, 3))
+```
+
+
+## Create table
+
+Format into pleasant output table using the 'gt' package. 
```

Comment:
This table didn't really render for me? I might also recommend looking into the kable and kableExtra R packages for creating pretty tables in Rmds.

Reply:
Thanks, will do when the final table is  needed!